python/cudf/cudf/core/index.py

```diff
@@ -2283,6 +2283,31 @@ def is_leap_year(self):
         res = is_leap_year(self._values).fillna(False)
         return cupy.asarray(res)
 
+    @property
+    def quarter(self):
+        """
+        Integer indicator for which quarter of the year the date belongs in.
+
+        There are 4 quarters in a year. With the first quarter being from
+        January - March, second quarter being April - June, third quarter
+        being July - September and fourth quarter being October - December.
+
+        Returns
+        -------
+        Int64Index
+        Integer indicating which quarter the date belongs to.
+
+        Examples
+        --------
+        >>> import cudf
+        >>> gIndex = cudf.DatetimeIndex(["2020-05-31 08:00:00",
+        ...    "1999-12-31 18:40:00"])
+        >>> gIndex.quarter
+        Int64Index([2, 4], dtype='int64')
+        """
+        res = extract_quarter(self._values)
+        return Int64Index(res, dtype="int64")
```

Comment:
Is it necessary to return an `int64`? The reason I'm asking is [1, 2, 3, 4] would easily fit in `int8` no?

Reply:
yes it would, but I think this was done to match Pandas behaviour.

Reply:
Would it be preferable here to use an `Int8Index` here instead?

Reply:
Yes, lets return `Int8Index` and ignore dtype checks in pytests